### ISSUE TITLE
Cancel action not behaving properly on Android

### ISSUE THREAD

**Issue by webberig** [auth0/auth0-cordova#33]
Consider this scenario:
- `client.authorize(...)` is called which opens a new browser window
- Instead of logging in, the user closes the window, returning back to the app

This is handled properly on iOS, an error with message "user canceled" is thrown and the app can act accordingly. However, on Android the callback is never fired, no error is being triggered. 

The browser is properly triggering a "Closed" event in `src/agent/browser`, but for some reason the auth0 client is only relaying the "user canceled" error for iOS:
https://github.com/auth0/auth0-cordova/blob/master/src/index.js#L96

TL;DR:
If a user on Android closes the popup browser without logging in, the app doesn't receive a callback.

**Comment by decates** [comment]
I'm getting this too, and it's obviously a pretty serious issue. Looking into a fix.

**Comment by decates** [comment]
As you said @webberig , the 'closed' event is only triggered for iOS when the user explicitly closes the browser window, however for Android, it gets triggered *before* the redirectUri comes through. As a 'solution', the closed event on non-iOS is ignored.  

Looking into it seems the only solution, without changes to InAppBrowser, or getting SafariViewController working on Android ([not yet](https://github.com/EddyVerbruggen/cordova-plugin-safariviewcontroller/issues/51), is to unfortunately introduce a time dependence/delay on Android - i.e. when Android says 'closed', wait a bit to see if a redirect comes through before triggering the `authorize` callback. 

Any thoughts @hzalaz ?

In any case I've got a workaround which I'll raise a PR for.

**Comment by brassier** [comment]
Any update on this one?  I have a similar need.
